cpu/nrf5x_common/periph/uart.c

```diff
@@ -275,6 +275,43 @@ void uart_poweroff(uart_t uart)
     dev(uart)->TASKS_STOPRX = 1;
 }
 
+int uart_mode(uart_t uart, uart_data_bits_t data_bits, uart_parity_t parity,
+              uart_stop_bits_t stop_bits)
+{
+
+    if (stop_bits != UART_STOP_BITS_1 && stop_bits != UART_STOP_BITS_2) {
+        return UART_NOMODE;
+    }
+
+    if (data_bits != UART_DATA_BITS_8) {
+        return UART_NOMODE;
+    }
+
+    if (parity != UART_PARITY_NONE && parity != UART_PARITY_EVEN) {
+        return UART_NOMODE;
+    }
+
+    uart_poweroff(uart);
+    dev(uart)->ENABLE = 0;
```

Comment:
Not sure if it is required to disable the controller before changing this. I put it in originally when things didn't work:tm: as expected. It is probably not required anymore (same with the enable below)

Reply:
Tested, powering down before and powering up after changing the mode is not required. Removed the code.